src/supervisor/watchers/handlers/pagination.ts

```diff
@@ -0,0 +1,92 @@
+import { V1ObjectMeta } from '@kubernetes/client-node';
+import { IncomingMessage } from 'http';
+import * as sleep from 'sleep-promise';
+
+import { calculateSleepSeconds } from '../../kuberenetes-api-wrappers';
+import { V1NamespacedList } from './types';
+import type { IRequestError } from '../../types';
+
+const PAGE_SIZE = 100;
+
+export async function paginatedList<
+  T extends { metadata?: V1ObjectMeta },
+  TList extends { items: Array<T> },
+>(
+  namespace: string,
+  list: TList,
+  listPromise: V1NamespacedList<TList>,
+): Promise<{
+  response: IncomingMessage;
+  body: TList;
+}> {
+  let continueToken: string | undefined = undefined;
+
+  const pretty = undefined;
+  const allowWatchBookmarks = undefined;
+  const fieldSelector = undefined;
+  const labelSelector = undefined;
+
+  let incomingMessage: IncomingMessage | undefined = undefined;
+
+  loop: while (true) {
+    try {
+      const listCall = await listPromise(
+        namespace,
+        pretty,
+        allowWatchBookmarks,
+        continueToken,
+        fieldSelector,
+        labelSelector,
+        PAGE_SIZE,
+      );
+      incomingMessage = listCall.response;
+
+      if (Array.isArray(listCall.body.items)) {
+        const trimmedItems = listCall.body.items.map((item: T) => {
+          delete item.metadata?.managedFields;
+          return item;
+        });
+        list.items.push(trimmedItems);
+      }
+
+      continueToken = listCall.body.metadata?._continue;
```

Comment:
what is a continue token? something that indicates there are more to list?

Reply:
Yes, something like "next page" starting from this token

Reply:
maybe it could be renamed to be clearer? or is that it's actual name?

Reply:
Yeah the parameter is named `continue`, but I'm open to suggestions!

Reply:
``shouldContinue = !!listCall.body.metadata?._continue;

Reply:
Ahh no, the return value of `_continue` actually a string, something like a token, hence it must be supplied as is. It's not really a flag/boolean!

Reply:
If you check out the documentation in VS Code, some of these parameters will start to make sense 😄 

<img width="517" alt="Screenshot 2021-11-03 at 10 33 46" src="https://user-images.githubusercontent.com/16717473/140045374-7c85cacc-8200-459c-8035-118887d70aab.png">